app.rb

```diff
@@ -0,0 +1,96 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each { |person| puts "ID: #{person.id}, #{person.class.name}: #{person.name}" }
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each_with_index { |book, index| puts "#{index}. #{book.title} by #{book.author}" }
+  end
+
+  def create_person
+    puts "Enter the person's name:"
+    name = gets.chomp
+
+    puts "Enter the person's age:"
+    age = gets.chomp.to_i
+
+    puts 'Is the person a student? (yes/no):'
+    is_student = gets.chomp.downcase == 'yes'
+
+    @people << (is_student ? Student.new(name, age) : Teacher.new(name, age))
+
+    puts "#{name} has been added as a #{is_student ? 'student' : 'teacher'}."
```

Comment:
- [ ] As per sneak peak video, I think you are supposed to add permission for student and specalization for teacher. This is currently absent in your app. See example where the options for student and teacher are 1 and 2 respectively.
```
 person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
```